File: Team1_Debateverse/mohitsingh/backend-springboot/src/main/java/com/server/debateverse/services/VoteService.java
package com.server.debateverse.services;

import java.util.List;

import com.server.debateverse.entities.Vote;
import com.server.debateverse.entities.VoteReq;

public interface VoteService {

    public String saveVotes(List<VoteReq> votes);

    public List<Vote> getVotes(Long userId, Long debateId);
    
}

File: Team1_Debateverse/mohitsingh/frontend-angular/src/app/services/vote-service/vote.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class VoteService {

  constructor(private _http:HttpClient) { }

  baseUrl: string = 'http://localhost:8080/api/votes';

  getVotesByUser(userId: any, debateId: any) {
    return this._http.get(`${this.baseUrl}/${userId}/${debateId}`);
  }

  vote(voteReq: any) {
    return this._http.put(`${this.baseUrl}/save`, voteReq, {responseType: 'text'});
  }
}
